core/src/main/java/org/radargun/stages/cache/background/AbstractLogLogic.java

```diff
@@ -219,28 +230,50 @@ protected void afterRollback() {
       delayedRemoves.clear();
    }
 
-   protected void afterCommit() {
+   protected boolean afterCommit() {
       boolean inTransaction = false;
       try {
+         Map<Long, Integer> delayedRemoveAttemptsMap = new HashMap<>();
          while (!stressor.isTerminated()) {
             try {
-               if (inTransaction) {
+               if (inTransaction || ongoingTx != null) {
                   try {
                      ongoingTx.rollback();
                   } catch (Exception e) {
-                     log.error("Failed to rollback ongoing transaction", e);
+                     log.error("Failed to rollback ongoing transaction.", e);
                   }
                }
                startTransaction();
                inTransaction = true;
                for (DelayedRemove delayedRemove : delayedRemoves.values()) {
-                  checkedRemoveValue(delayedRemove.keyId, delayedRemove.oldValue);
+                  try {
```

Comment:
Delayed remove can fail during commit, too, and you don't count commit failures. Besides that, it seems to me that counting this for individual keys is rather overkill, I'd use single counter for the whole transaction (if one remove fails, whole transaction needs to be rolled back anyway).